src/SignalR/perf/benchmarkapps/Crankier/Server/ConnectionCounterHostedService.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Crankier.Server
+{
+    public class ConnectionCounterHostedService : IHostedService, IDisposable
+    {
+        private Stopwatch _timeSinceFirstConnection;
+        private readonly ConnectionCounter _counter;
+        private ConnectionSummary _lastSummary;
+        private Timer _timer;
+
+        public ConnectionCounterHostedService(ConnectionCounter counter)
+        {
+            _counter = counter;
+            _timeSinceFirstConnection = new Stopwatch();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _timer = new Timer(DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(1));
+            
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
```

Comment:
It is possible to have DoWork calls overlap with a one-second-interval timer invoking it in certain scenarios (like thread-starved scenarios), so you might want to guard against that. You can copy the logic from https://github.com/aspnet/AspNetCore/blob/8c02467b4a218df3b1b0a69bceb50f5b64f482b1/src/Servers/Kestrel/Core/src/Internal/Infrastructure/Heartbeat.cs#L31-L72.

Reply:
done